account_payment_order/models/account_move.py

```diff
@@ -48,7 +48,7 @@ def _get_payment_order_communication(self):
             if (self.reference_type or "none") != "none":
                 communication = self.ref
             elif self.is_purchase_document():
-                communication = self.ref or self.payment_reference
+                communication = self.payment_reference or self.ref or ""
```

Comment:
Do you really need to add `or ""`?

Reply:
I added `or ""` because, in version 16, the **__get_payment_order_communication_** function is divided into two separate methods: _**_get_payment_order_communication_direct**_ and _**_get_payment_order_communication_full**_.

If I remove the `or ""`, the line `communication += " " + " ".join(references)` throws the error `unsupported operand type(s) for +=: 'bool' and 'str'`, as self.ref can be False. To avoid this issue, we can either split the function as in version 16 or handle it with `or ""` to ensure communication is always a string.